lib/Dialect/Seq/SeqOps.cpp

```diff
@@ -556,6 +568,10 @@ LogicalResult FirRegOp::verify() {
     if (getIsAsync())
       return emitOpError("register with no reset cannot be async");
   }
+  if (auto preset = getPresetAttr()) {
+    if (preset.getType() != getType())
+      return emitOpError("preset type must match register type");
+  }
```

Comment:
This error path would benefit from a test.

This implies that signed reset values are not supported unless converted to the proper number format for the register. I think this is totally fine.